include/oneapi/dpl/pstl/hetero/memory_ranges_impl_hetero.h

```diff
@@ -0,0 +1,244 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_MEMORY_RANGES_IMPL_HETERO_H
+#define _ONEDPL_MEMORY_RANGES_IMPL_HETERO_H
+
+#if _ONEDPL_CPP20_RANGES_PRESENT
+
+#    include <ranges>
+#    include <utility>
+#    include <cassert>
+#    include <functional>
+#    include <type_traits>
+
+#    include "algorithm_ranges_impl_hetero.h" // for __pattern_walk_n
+#    include "dpcpp/utils_ranges_sycl.h"      // for oneapi::dpl::__internal::__ranges::views::all, etc.
+#    include "dpcpp/execution_sycl_defs.h"    // for __hetero_tag
+
+namespace oneapi
+{
+namespace dpl
+{
+namespace __internal
+{
+namespace __ranges
+{
+
+//---------------------------------------------------------------------------------------------------------------------
+// pattern_uninitialized_default_construct
+//---------------------------------------------------------------------------------------------------------------------
+
+template <typename _BackendTag, typename _ExecutionPolicy, typename _R>
+std::ranges::borrowed_iterator_t<_R>
+__pattern_uninitialized_default_construct(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _R&& __r)
+{
+    using _ValueType = typename std::ranges::range_value_t<_R>;
```

Comment:
```suggestion
    using _ValueType = std::ranges::range_value_t<_R>;
```

This `typename` looks really weird. This is the ubiquitous comment

Reply:
Removed here and in other places